### DISCUSSION TITLE
Changing Peclet and Damköhler Numbers for Simulating Dissolution Regimes Using reactiveTransportDBSFoam

### DISCUSSION THREAD

**Discussion by Yeyang28** [GeoChemFoam/GeoChemFoam#30] [Q&A] [answered]
Dear Professor Hannah,

First, thank you for sharing code and knowledge with us. 

I am currently engaged in simulating various Peclet and Damköhler numbers to explore different dissolution regimes using reactiveTransportDBSFoam. I have had success in simulating multiple cylinders within wormhole and compact dissolution regimes. However, I am encountering challenges with the uniform dissolution regime. I achieved accurate velocity profiles, reaction rate profiles, and porosity profiles. However, I observed similar acid concentrations inside and outside the mineral cylinders, the concentration gradient across the entire domain is super small. The concentration value inside the cylinder is almost equal to the inlet concentration. The porosity I set for solid porosity is 0.0001, and the mesh size is 0.02 mm.

Could you tell me how to address this issue ? Thank you for your time and assistance.

Best
Yeyang

**Comment by GeoChemFoam** [comment]
Hi Yeyang,

Sounds like a meshing or boundary condition problem. Have you rendered it in Paraview to check?

Cheers,

Hannah

From: yeyang ***@***.***>
Sent: Friday, November 17, 2023 12:04 PM
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GeoChemFoam/GeoChemFoam] Changing Peclet and Damköhler Numbers for Simulating Dissolution Regimes Using reactiveTransportDBSFoam (Discussion #30)

****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Dear Professor Hannah,

First, thank you for sharing code and knowledge with us.

I am currently engaged in simulating various Peclet and Damköhler numbers to explore different dissolution regimes using reactiveTransportDBSFoam. I have had success in simulating multiple cylinders within wormhole and compact dissolution regimes. However, I am encountering challenges with the uniform dissolution regime. I achieved accurate velocity profiles, reaction rate profiles, and porosity profiles. However, I observed similar acid concentrations inside and outside the mineral cylinders, the concentration gradient across the entire domain is super small. The concentration value inside the cylinder is almost equal to the inlet concentration. The porosity I set for solid porosity is 0.0001, and the mesh size is 0.02 mm.

Could you tell me how to address this issue ? Thank you for your time and assistance.

Best
Yeyang

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/30>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFQXX5IFP2PWMDEV6NLYE5HBXAVCNFSM6AAAAAA7PV3PQKVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVHA3DCNZUGA>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.******@***.***>>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Reply by Yeyang28** [reply]
Hi,

Thanks for your fast reply!
The concentration profile (with 200s) in paraView is like the attachment photo. The cylinder is not close to the boundary. Fluid with porosity 1, solid with porosity 0.0001. And no matter how large the K_0 (KC constant), the concentration gradient is always small. I tried "iVos" and "Vos-psi"; the problems are the same. 
For porosity, it indeed dissolves homogeneously. Just the concentration profile is not correct.

![Conc  with 200s ](https://github.com/GeoChemFoam/GeoChemFoam/assets/120320269/ec8f8780-6d33-4b3a-a728-96c4424e7f4a)

Best
Yeyang

**Comment by GeoChemFoam** [comment]
I don’t know from these pictures. My advice is to start with the tutorial as is and then change only a single parameter at a time to troubleshoot (i.e. first the image, then the resolution, then flowrate, reaction constant, etc).

**Reply by Yeyang28** [reply]
Hi. Professor,

I have checked the single parameters. After testing, I think the reason is the value of the reaction constant. If the reaction constant is small, it makes the concentration gradient super small. In the code of transport equation, the reaction term "R" does not depend on the concentration. Therefore, the lower reaction constant will give a small reaction term "R" and result in a similar concentration gradient between the inside and outside cylinder. Am I correct?

Best
Yeyang

**Comment by Yeyang28** [comment]
Dear Professor Hannah

Upon reviewing the tutorial, I encountered an issue that I believe may be related to the concentration gradient. Specifically, I experimented by setting the reaction constant to zero (kreac=0) in the "initCase.sh" file, with the expectation that the mineral would not dissolve under these conditions, and a clear concentration gradient could be shown between the inside and outside of the cylinder. However, the concentration gradient depicted in the attached photo (labeled "Conc. with 800s") does not exhibit any variance (further time steps are all similar conc. gradient behavior). This result is my confusion.

I could easily simulate wormhole and compact dissolution regimes in the previous. But when it comes to the uniform dissolution, I decrease the reaction constant (kreac), and the concentration gradient between the inside and outside of the cylinder is super small.

I would greatly appreciate your insight into whether this outcome aligns with the expected behavior of the model. Thank you very much for your time and assistance! 

![Concentration_800s](https://github.com/GeoChemFoam/GeoChemFoam/assets/120320269/0b940971-3d3a-4460-b6d9-2f720d79b847)


Best regards
Yeyang

**Reply by julienmaes** [reply, marked as answer]
C is the concentration in the pore-space. In the model you cannot have eps=0 (this is a limitation of the method). Instead, the solid is modelled as a porous media with small porosity (eps=0.0001 for example) and small permeabiltiy. In your case, since kreac=0, the concentration will be C=1.3e-2 everywhere, included in the 0.0001 pores in the the solid, and that's what you're seeing
